lib/gears/matcher.lua

```diff
@@ -0,0 +1,292 @@
+--- A module to build a set of properties based on a graph of rules.
+--
+-- Rules come from a graph of sources. Each source depend on other and must be
+-- applied before other sources. For this module to work, the graph needs to be
+-- solvable.
+--
+-- @DOC_text_gears_matcher_default_EXAMPLE@
+--
+-- @author Julien Danjou &lt;julien@danjou.info&gt;
+-- @copyright 2009 Julien Danjou
```

Comment:
Update maybe?!

Reply:
I didn't write this code. It's cut/pasted from `awful.rules`. The only part I wrote is 3/4 of the graph source method.